vault/wallet.go

```diff
@@ -256,84 +125,74 @@ func (w *WalletImpl) Sign(txn *transaction.Transaction) error {
 	return nil
 }
 
-func verifyPasswordKey(passwordKey []byte, passwordHash []byte) bool {
-	keyHash := sha256.Sum256(passwordKey)
-	if !bytes.Equal(passwordHash, keyHash[:]) {
-		fmt.Println("error: password wrong")
-		return false
-	}
-
-	return true
+func (w *Wallet) VerifyPassword(password []byte) error {
+	return w.WalletStore.WalletData.VerifyPassword(password)
 }
 
-func (w *WalletImpl) ChangePassword(oldPassword []byte, newPassword []byte) bool {
-	// check original password
-	oldPasswordKey := crypto.ToAesKey(oldPassword)
-	passwordKeyHash, err := HexStringToBytes(w.Data.PasswordHash)
+func (w *Wallet) ChangePassword(oldPassword, newPassword []byte) error {
+	account, err := w.DecryptAccount(oldPassword)
+	if err != nil {
+		return err
+	}
+
+	address, err := account.ProgramHash.ToAddress()
```

Comment:
Maybe accout here didn't decrypted correctly, failed to change password with an old version of wallet file when using new nknc client

Reply:
The only possibility that wallet decrypt failed is because the wallet file is modified and corrupted, and we can do nothing about it.